src/open3DTool/planeUtils.py

```diff
@@ -0,0 +1,66 @@
+from src.algorithmForNN.pointCloudUtils import (
+    convert_point_cloud_to_numpy_array,
+)
+
+import open3d as o3d
+import numpy as np
+
+
+def pick_points_utils(point_cloud: o3d.geometry.PointCloud):
+    picked_visualizer = o3d.visualization.VisualizerWithEditing()
+    picked_visualizer.create_window()
+    picked_visualizer.add_geometry(point_cloud)
+    picked_visualizer.run()
+    picked_visualizer.destroy_window()
+    return picked_visualizer.get_picked_points()
+
+
+def get_distance_to_all_points(
+    point_cloud: o3d.geometry.PointCloud, plane: np.ndarray
+) -> np.ndarray:
+    numpy_point_cloud = np.asarray(point_cloud.points)
+    ones_array = np.ones((numpy_point_cloud.shape[0], 1), dtype=np.float64)
+    numpy_point_cloud = np.append(numpy_point_cloud, ones_array, axis=1)
+    plane = plane.T
+
+    distances = np.abs(numpy_point_cloud @ plane) / np.linalg.norm(plane)
+
+    return distances
+
+
+def get_indexes_of_points_on_plane(
+    distances: np.ndarray, plane_distance: np.float64
+) -> np.ndarray:
+    return np.where(distances <= plane_distance)[0]
+
+
+def get_plane_using_SVD(points: np.ndarray) -> np.ndarray:
+    centroid = points.mean(axis=0)
+    points_temp = points - centroid
+    _, _, V_T = np.linalg.svd(points_temp)
+    normal_vector = V_T[2]
+    normal_vector = np.append(normal_vector, -np.dot(normal_vector, points[0]))
```

Comment:
Why do you take first point from the points array and calculate this strange `dot`? Definitely because of noise nature different points will give different results whereas the solution is single.

Reply:
My idea was that the first point is the "most correct" one. How else can I calculate the coefficient `d`?

Reply:
The minimum value will be achieved in centroid point, I suppose.

Reply:
OK